### PR TITLE
adds colony self destruct mechanisms to big red, desert dam and lv

### PR DESCRIPTION
**Pull request by AndroBetel**
# About the pull request

adds self-destruct mechanisms to big red ETA, desert dam lab and LV lab (LV one is a rare nightmare insert)
they are activated using nuclear disks, which are made unacidable
after two minutes, they explode with explosion power similar to HE OB

image

# Explain why it's good for the game

this pr was inspired by battlefield levolution system, as it allows players to single-handedly change entire locations (not without some set-up, of course)
i find map interactivity in CM lacking, so this pr also adds said interactivity to some maps

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Adds colony self-destruct mechanisms to Big Red, LV-624 and Desert Dam, activated via nuclear disks.
/:cl:

### CODE REVIEW

File: code/game/machinery/colony_selfdestruct.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		icon_state = "selfdestruct2"
+
+	if(timeleft < stage_2 && timeleft > stage_3)
+		icon_state = "selfdestruct3"
+
+	if(timeleft < stage_3 && timeleft > 0)
+		icon_state = "selfdestruct4"
+
+	if(timeleft <= 0)
+		icon_state = "selfdestructboom"
```
Comment (kiVts) [inline code/game/machinery/colony_selfdestruct.dm:84]:
```suggestion

	if(timeleft <= 0)
		icon_state = "selfdestructboom"
		return
		
	if(timeleft < stage_3)
		icon_state = "selfdestruct4"
		return
		
	if(timeleft < stage_2)
		icon_state = "selfdestruct3"
		return
		
	if(timeleft < stage_1)
		icon_state = "selfdestruct2"
		return
		
	if(timeleft < initial(timeleft))
		icon_state = "selfdestruct1"
```

this looks generally less confusing, dont you think? 

also could be else if spam

File: code/game/machinery/colony_selfdestruct.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	desc = "Weyland-Yutani produced SelDest-15 low-yield nuclear bomb. Guaranteed to destroy all corporate secrets along with personnel or your money back!"
+	icon_state = "selfdestruct"
+	density = TRUE
+	unslashable = TRUE
+	unacidable = TRUE
+	use_power = USE_POWER_NONE
+	var/obj/item/disk/nuclear/dat_disk
+	var/timeleft = 2 MINUTES
+	var/explosiontime
+	var/explosionpower = 1200
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:12]: cod edoc

File: code/game/machinery/colony_selfdestruct.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+/obj/structure/machinery/colony_selfdestruct/attack_hand(mob/user)
+	if(user.is_mob_incapacitated() || get_dist(src, user) > 1 || isRemoteControlling(user))
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:15]: i think we have an in_range proc

File: code/game/machinery/colony_selfdestruct.dm [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		return
+
+	if(!dat_disk)
+		to_chat(user, SPAN_WARNING("[src] can't be activated without the nuclear disk!"))
+		return
+	else
+		user.visible_message(SPAN_WARNING("[user] starts to pull the disk out of [src]..."), SPAN_WARNING("You start to pull the disk out of [src]... you hope it's not stuck."))
+		if(do_after(user, 150 * user.get_skill_duration_multiplier(SKILL_ENGINEER), INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:23]: SECONDS macro

File: code/game/machinery/colony_selfdestruct.dm [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+			disable()
+
+/obj/structure/machinery/colony_selfdestruct/proc/enable()
+	playsound(loc, 'sound/items/Deconstruct.ogg', 100, 1)
+	explosiontime = world.time + timeleft
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:28]: use a timer

File: code/game/machinery/colony_selfdestruct.dm [hunk 6]
```diff
@@ -0,0 +1,95 @@
+/obj/structure/machinery/colony_selfdestruct
+	name = "self-destruct mechanism"
+	desc = "Weyland-Yutani produced SelDest-15 low-yield nuclear bomb. Guaranteed to destroy all corporate secrets along with personnel or your money back!"
+	icon_state = "selfdestruct"
+	density = TRUE
+	unslashable = TRUE
+	unacidable = TRUE
+	use_power = USE_POWER_NONE
+	var/obj/item/disk/nuclear/dat_disk
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:9]: null on destroy

File: code/game/machinery/colony_selfdestruct.dm [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+	if(timeleft < stage_3 && timeleft > 0)
+		icon_state = "selfdestruct4"
+
+	if(timeleft <= 0)
+		icon_state = "selfdestructboom"
+
+/obj/structure/machinery/colony_selfdestruct/attackby(obj/item/weapon, mob/living/user)
+	if(istype(weapon, /obj/item/disk/nuclear) && !dat_disk)
+		user.visible_message(SPAN_WARNING("[user] begins to put [weapon] inside [src]..."), SPAN_WARNING("You begin to put [weapon] inside [src]. Ooh boy..."))
+		if(do_after(user, 150 * user.get_skill_duration_multiplier(SKILL_ENGINEER), INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
```
Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:89]: seconds macro

Comment (Zonespace27) [inline code/game/machinery/colony_selfdestruct.dm:89]: early return

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by Nanu308** [comment]
Fix up your PR and resolve prior reviews, and I'll map review it.

File: maps/map_files/LV624/science/10.selfdestruct.dmm
Comment (Nanu308) [inline maps/map_files/LV624/science/10.selfdestruct.dmm:1]: mapping lgtm

File: maps/map_files/DesertDam/Desert_Dam.dmm
Comment (Nanu308) [inline maps/map_files/DesertDam/Desert_Dam.dmm:1]: mapping lgtm

File: maps/map_files/BigRed/BigRed.dmm
Comment (Nanu308) [inline maps/map_files/BigRed/BigRed.dmm:1]: mapping lgtm

**Review by Nanu308 — COMMENTED** [review]
mapping lgtm

**Pull request closed without merging**